Use the official docker images while building hygdrop

theanalyst opened the pull request:
Changes Dockerfile to use the hylang image instead of building hy
itself, also requirements file modified to remove earlier dependancies
of hy & astor

Ping @tianon @paultag @copyninja for review

theanalyst committed:
Use the official docker images while building hygdrop

Changes Dockerfile to use the hylang image instead of building hy
itself, also requirements file modified to remove earlier dependancies
of hy & astor

tianon commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+
+ADD . /opt/hygdrop
+WORKDIR /opt/hygdrop
+RUN pip3 install -r requirements-docker.txt
```
Why not do something like:

``` Dockerfile
COPY requirements-docker.txt /opt/hygdrop/
RUN pip3 install -r requirements-docker.txt
COPY . /opt/hygdrop
```

That way we get to utilize the Docker build cache a little better.

theanalyst replied:
Ah, Didn't think of that.thanks! changing in a bit

tianon replied:
Also, since the `hylang` image already has Hy installed, can't we use the same `requirements.txt` and let the already-installed Hy satisfy the Hy dep?

theanalyst replied:
Yep.. that is why created `requirements-docker` which doesn't include hy.. the plain `requirements` contains hy if you're doing it the traditional way. WDYT?

tianon replied:
No, I mean to say that you don't need to split them, do you?  Isn't `pip` smart enough to notice that the `hy>=0.10` dep is already satisfied and skip it appropriately?

theanalyst replied:
ah.. stupid me :)

tianon replied:
Just to make sure I verified that assumption:

``` console
root@5006d933f5b6:/usr/src/python# pip install hy
Requirement already satisfied (use --upgrade to upgrade): hy in /opt/hylang/hy
...
```

:smile:

theanalyst committed:
Better dockerfile instructions

- made installing from requirements first to utilize build cache better
- removed redundant requirements

theanalyst commented:
@tianon added the steps you said (& removed unnessary requirements file), let me know if this is ok

tianon commented:
LGTM
